### PR TITLE
Refactor #9422 [v98] Header and footer in BVC

### PR DESCRIPTION
**Pull request by lmarceau**
This is the base work for the rest of the #9422 feature. The goal was to streamline how we manage header and footer views with BVC, removing edge cases and making it possible to easily put a subview from the header into the footer.

## Key changes:
- Footer and header are now stackviews. This removes extra constraints we needed to do manually.
- Created a stack view extension so it's easy to animate adding and removing stack views, making sure at the same time the view is remove properly.
- Reader mode is part of the header. This was made to ensure it can also be added in the footer (for #9422)
- We now have a AlphaDimmable protocol so TabScrollController doesn't have to keep a reference to urlBar and readerMode to change their alphas.
- Fixed indentation and nits, also restructuring the code to remove arrow code in commit `95428b5`
#9946 

TLDR; this is a code refactor and isn't supposed to induce any change whatsoever in how the UI is rendered.

### CODE REVIEW

**Comment by lmarceau** [comment]
Updated to point to epic branch. Found an issue with reader mode in this branch. Will be resolved in part 2 of this feature. Keeping this PR opened to avoid having a huge PR at the end.

**Review by nbhasin2 — APPROVED** [review]

File: Client/Frontend/Browser/TopTabsViewController.swift [hunk 1]
```diff
@@ -133,8 +132,12 @@ class TopTabsViewController: UIViewController {
...
         // Setup UIDropInteraction to handle dragging and dropping
         // links onto the "New Tab" button.
-            let dropInteraction = UIDropInteraction(delegate: topTabDisplayManager)
-            newTab.addInteraction(dropInteraction)
+        let dropInteraction = UIDropInteraction(delegate: topTabDisplayManager)
+        newTab.addInteraction(dropInteraction)
+
+        view.snp.makeConstraints { make in
+            make.height.equalTo(TopTabsUX.TopTabsViewHeight)
+        }
```
Comment (nbhasin2) [inline Client/Frontend/Browser/TopTabsViewController.swift:140]: nit: would be nice to make it not use snapkit in future 😅

**Comment by lmarceau** [comment]
I'll investigate snapkit removal in next PRs for this epic branch 💪

File: Client/UIStackViewExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+extension UIStackView {
+
+    func toggleStackViewVisibility(show: Bool) {
```
Comment (nbhasin2) [inline Client/UIStackViewExtensions.swift:9]: nit: to error proof some of these extension functions

Comment (lmarceau) [reply Client/UIStackViewExtensions.swift:9]: Will do some tests in next PR as well
